### DISCUSSION TITLE
Uncaught TypeError: jQuery.jmesa.setPage is not a function

### DISCUSSION THREAD

**Discussion by Zaquer** [yujiaao/jmesa#2] [Q&A]
Hello!

I wanted to ask you something about this new version of jmesa. I will try to explain the problem the best I can: 
I have a project which was made on java 8 with jmesa 3.0.4, and is now upgrading to java 21 with jakarta, and a lot of other libraries... When updating jmesa, for which I thank you a lot, this repository saved me a lot of hours of changing the tables, the java part was fast, almost everything worked without needs of changing, except the toolbar. I remade the toolbar we had with the new SimpleToolbar included in the repository, and it works almost perfect, except for the botons to change the page of the table. I have tow scenaries:
If I don't change the jmesa.js I had, which are the old ones from the jmesa 3.0.4, the following errors appear:

 **Uncaught TypeError: jQuery.jmesa.setPage is not a function**
 
 It doesn't say anything else.  I thought that maybe the javascript files where also updated, so I came here, download the new ones , implemented them, and the error just disappear... but it wasn't working... I checked the files and notice that the ones are shorter than the old ones, which has more functions; and some of the functions that the old jmesa used to change pages on a table, seems to not be in the new javascript files. So, my cuestion is, is this intended? shouldn't the functions needed for jmesa be already there or do we have to do our own functions? Is there something I'm missing? 
 
 This is an example of ones of the functions I think is missing in the new version of jquery.jmesa.js (I think this is the one that made the old jmesa toolbar buttons work, I have worked a lot with jmesa so I'm not sure, but from what I have saw, it has to be something like this if not this one.) :
 `createHiddenInputFields : function(form) {
            var limit = this.limit;

            var exists = $(form).find(':hidden[name=' + limit.id + '_p_]').val();
            if (exists) {
                return false;
            }

            if (this.worksheet.save) {
                $(form).append('<input type="hidden" name="' + limit.id + '_sw_" value="true"/>');
            }

            if (this.worksheet.filter) {
                $(form).append('<input type="hidden" name="' + limit.id + '_fw_" value="true"/>');
            }

            if (this.worksheet.clear) {
                $(form).append('<input type="hidden" name="' + limit.id + '_cw_" value="true"/>');
            }

            if (this.worksheet.addRow) {
                $(form).append('<input type="hidden" name="' + limit.id + '_awr_" value="true"/>');
            }

            if (this.worksheet.removeRow) {
                $.each(this.worksheet.removeRow, function(key, value) {
                    $(form).append('<input type="hidden" name="' + limit.id + '_rwr_" value="' + value + '"/>');
                });
            }

            /* tip the API off that in the loop of working with the table */
            $(form).append('<input type="hidden" name="' + limit.id + '_tr_" value="true"/>');

            /* the current page */
            $(form).append('<input type="hidden" name="' + limit.id + '_p_" value="' + limit.page + '"/>');
            $(form).append('<input type="hidden" name="' + limit.id + '_mr_" value="' + limit.maxRows + '"/>');

            /* the sort objects */
            var sortSet = limit.getSortSet();
            $.each(sortSet, function(index, sort) {
                $(form).append('<input type="hidden" name="' + limit.id + '_s_'  + sort.position + '_' + sort.property + '" value="' + sort.order + '"/>');
            });

            /* the filter objects */
            var filterSet = limit.getFilterSet();
            $.each(filterSet, function(index, filter) {
                $(form).append('<input type="hidden" name="' + limit.id + '_f_' + filter.property + '" value="' + filter.value + '"/>');
            });

            return true;
        },`
 
 Thanks for your time, and sorry for the long text!

**Comment by yujiaao** [comment]
I change jmesa java part only for my personal property, may be not suitable for you, and the `javax` to `jakarta` migration is only 'compilable',  the `preferences` may be not working. 

I 'm not care about the jQuery part because in my project i integrated with React so jQuery plugin not maintained.
